src/NServiceBus.Raw/StartableRawEndpoint.cs

```diff
@@ -32,14 +32,27 @@ public async Task<IReceivingRawEndpoint> Start()
             }
 
             var runningInstance = new RunningRawEndpointInstance(settings, receiver, transportInfrastructure, dispatcher, SubscriptionManager, localAddress);
+
             // set the started endpoint on CriticalError to pass the endpoint to the critical error action
             criticalError.SetEndpoint(runningInstance);
 
-            if (receiver != null)
+            try
```

Comment:
Why do we need this code? The receiver returned by `CreateReceiver` cannot be null and the `criticalError` is already initialized. I think that the changes to the `RawTransportReceiver` should be enough to fix the RabbitMQ issue.

Reply:
![connection-leak](https://user-images.githubusercontent.com/83468000/142486757-6bb88581-d28a-4eba-8caa-ff99d81de847.gif)

I recorded this gif to show the connection leak if that stop call is commented out.  The critical error callback wasn't called in my test project, it's possible I misconfigured it?  You can see that code in the gif (line 25).  At least for RabbitMQ, the endpoint needs to be specifically stopped for it to clean up its connection.

Reply:
OK. Found it. In RabbitMQ it is the TransportInfrastructure object that holds these connections and the transport infrastructure is stopped only when the `StoppableRawEndpoint` is stopped. 

I must admit that the lifecycle of the Raw endpoint is probably too complex:
 - `Initialize` -> after that you can send messages
 - `Start` -> after that the receive pump works
 - `StopReceiving` -> after that the receive pump does not work but you can still send messages
 - `Stop` -> completely disposed
 
What is worse there is this shortcut method `RawEndpoint.Start` that calls `Initialize` and `Start`.

I think that the `RawTransportReceiver` change is OK in the sense that if `Start` on the `IPushMessages` fails we call stop to ensure that the `Start` did not fail mid way through. Although one could argue that the `Start` method on `IPushMessages` should be written in such a way that when it throws, it ensures that the state is clean.

But the bigger issue is the `Start` method. If the user calls `RawEndpoint.Create` and gets an exception when starting the endpoint, can they re-try starting the same endpoint or is its state corrupted? If it is then what to do with it?

In the current state the user needs to call `Stop` after a failed start to dispose off resources but if they used the shortcut method there is no instance to call `Stop` on. 

If I understand correctly with your change the failure to start basically stops the endpoint and renders it unusable. That sounds like a good option.